### ISSUE TITLE
Model Details

### ISSUE THREAD

**Issue by Quinn-Donnelly** [Gods-Of-ACM/backEnd#41]
Implement route to send the details of a model. This will send the attributes and types to the caller. Proposed structure is and array with objects holding name, type, and a Boolean that will indicated that the attribute is a foreign relation and a string that will be set to that model's name. The names of the types will match type names from sequelize API.

```json
[{
    "name": "substance_id",
    "type": "INTEGER",
    "isForeignKey": false,
    "foreignModelName": ""
}]
```

**Comment by Quinn-Donnelly** [comment]
@wrejcodes what do you think of this structure for the response is there any other data they might need.

**Comment by wrejcodes** [comment]
Is this to get the information about a model to the frontend?

**Comment by Quinn-Donnelly** [comment]
Yes this would be the front end sending a request and the response would contain the above in the payload. It is information that will be needed for the admin page in the assignment. Figured this would be better than hardcoding the frontend.

**Comment by wrejcodes** [comment]
Ah yes that’s where I was confused. I thought the front end was sending the
proposed object and I was really confused why they were sending the details
to Get the details back. This makes more sense haha. I think this would be
fine as long as we are only issuing the data request the first time they
request the details. I could see a situation where someone is looking
between two tables deciding which to pick and making a network request each
time would not be desired.


One addition piece of information about a field could be validation info.
More specifically we could give information about what we want to check on
front end as far as form submission goes.

**Comment by Quinn-Donnelly** [comment]
So long as the front end checks their redux store before making the request and seeing they have the information already (they could also launch it each time they switch however due to the way redux works it would immediately show the data using the store and update if the data changes)

Once we have validations i think that would be a good addition. I am not aware though of any validations that need to be run other than type and relations. I could be forgetting something though.

**Comment by wrejcodes** [comment]
Yeah I don’t think we do right now but I think it would be useful in the future! I think this looks good though

**Comment by Quinn-Donnelly** [comment]
I think for now since the focus is on other tasks the detail will not return the relations as of yet as there are no relations in the sequelize model defined. Once they are added I will add them to the response. 

Proposed response as stands now
```json
[{
    "name": "substance_id",
    "type": "INTEGER",
    "isDefault": false
}]
```
The is default will simply indicate if the field is included as one of the default fields of search and interest.

**Comment by wrejcodes** [comment]
I think that sounds like a good idea. At this point extra work should be
avoided until we have the requirements met

**Comment by Quinn-Donnelly** [comment]
What route should this be? GET /<huh>/<oh yeah>

**Comment by Quinn-Donnelly** [comment]
that markdown did some weird things to my sample url.

**Comment by wrejcodes** [comment]
/api/details?model=<modelname>

Maybe something like that

**Comment by Quinn-Donnelly** [comment]
I have something similar currently with /api/models?name=< modelname >

**Comment by wrejcodes** [comment]
Yeah I think that works. Looks good to me!

**Comment by Quinn-Donnelly** [comment]
Implemented but will await #45 to merge to use the new require routing.

**Comment by Quinn-Donnelly** [comment]
Resolved with #46

**Comment by Quinn-Donnelly** [comment]
Which hasn't merged reopening.
